### PR TITLE
A resolved note shows the agent's line before it leaves, and a tab wraps where it draws

### PR DESCRIPTION
**Pull request by breferrari**
Four edge cases on the note surface, found by reading `vigia mcp` and the store against B21 rather than by hitting them. Each is a failing test first, then the fix. The third moves a ruling and takes `SPEC.md` with it in the same commit.

## The server reads `all` as a boolean

`all` was read through `as_bool().unwrap_or(false)`, and `false` is the branch that deletes. A client sending the schema's boolean as a string, which nothing on the way in rejected, asked to read the resolved notes and had them removed instead. It is the one argument on this surface whose default destroys, so a value that is neither true nor false is refused with a sentence now. Absent is still the default.

## The hook keeps quiet about a session it cannot name

`hooked` handed the registry whatever `session_id` the payload carried, and an id the registry cannot name a file after came back as an error that `register` printed to stderr and exited non-zero for. That hook runs at the start and end of every session in every project the reader opens, and its own rule is that only a write it attempted and could not finish is reported. `vigia_core::names_a_record` lets a caller ask before it tries. Session ids are lowercase UUIDs today, so this one is latent.

## The pane takes a resolved note's file, and no listing does

This is the one worth reading closely.

B21 has the agent resolve "with a line the reader watches arrive before the note leaves". The server pruned a resolved file on the agent's next listing, which is not the same moment. Resolve-then-list in one breath is the ordinary shape of an agent working a queue, and both can land inside a single wake: the pane then never meets `Resolved` at all, only a note that was there and is not, which is a withdrawal. The rows dissolve still showing the reader's own body and the word `seen`, and the line the tool makes required is never drawn. With no pane open while the agent worked, the file was gone before the reader could start one, so the spec's own promise that a resolved note met at startup departs the same way could not be kept either.

Nothing reachable from the server can tell a note the reader watched leave from one resolved into an empty room. The pane can, because it is the thing that drew it. A listing now passes over a resolved note and leaves the file; `Ledger::settle` returns the ids whose departure has run; the shell removes them on the frame that drops the rows. A refusal is one footer alert on the same terms as the other four.

`SPEC.md` §11.2 moves with the code rather than after it, and three statements elsewhere that claimed the old ownership are corrected, including the B21 build note's claim that nothing accumulates because the next listing prunes.

## A tab in a note wraps as the columns it is drawn in

The wrap measured a paragraph two ways. The loop deciding whether to break used `width_of`, which is unicode width, where a tab is the nothing the buffer draws it as. The cut came from `split_at`, which walks a tab out to its stop. Forty drawn columns took three rows instead of two, half of each row went unused, and the last carried nothing but the status word. `render::detabbed` expands tabs before anything measures, so the wrap, the cut and the draw see one string, and pasted indentation is visible rather than silently gone. The box detabs on paste too, since `Tab` there is already a soft indent and a literal one stood the caret a column off its character.

## On the written layer

The spec change first ran 268 bytes over, and every ceiling equals its file exactly, so there was nothing to hand over. Cutting the rationale down to the rule, and dropping a sentence at line 1495 that §11.2 now states, brought `SPEC.md` to **391062, nine bytes under where it started**. The ceiling falls rather than rises and no budget was raised.

Note this touches `WRITTEN_LAYER_BUDGET` in `crates/vigia/tests/package.rs`, which #453 also edits.

## Verification

`cargo test --workspace --no-fail-fast`: 1408 passed, 0 failed. `cargo fmt --check` clean. `RUSTFLAGS=-D warnings cargo clippy --workspace --all-targets` clean. Each of the three commits was formatted and tested before the next was applied.

Not covered here, and left as findings rather than changes: the resource read is still a mutating operation, `list_changed` is still used where the spec's own `resources/updated` would fit, and neither reply nor note body has a length cap.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/vigia/src/mcp.rs [hunk 1]
```diff
@@ -199,7 +199,21 @@ impl Server {
...
+                    None | Some(Value::Null) => false,
+                    Some(Value::Bool(all)) => *all,
+                    Some(other) => {
+                        return Ok(failed(&format!(
+                            "notes takes all as a boolean and {other} is not one; leave it out \
+                             to list the open notes and prune the resolved ones, or pass true to \
+                             read the resolved ones and prune nothing"
+                        )));
+                    }
+                };
```
[the remark this answers is not in the document]
Comment (breferrari) [reply crates/vigia/src/mcp.rs:215]: Fixed in 59617db. You are right that it described a behaviour the next commit removed: commit 1 wrote that sentence while listings still pruned, and commit 2 moved the prune to the pane without revisiting it. It now reads "leave it out to list the open notes, or pass true to list the resolved ones too".

File: crates/vigia/src/view.rs [hunk 2]
```diff
@@ -470,9 +470,11 @@ struct Pin {
 fn prose_rows(text: &str, room: usize) -> Vec<String> {
     text.split('\n')
         .flat_map(|paragraph| {
-            prose_pieces(paragraph, room)
+            let paragraph = crate::render::detabbed(paragraph);
+            prose_pieces(&paragraph, room)
                 .into_iter()
                 .map(|piece| paragraph[piece].to_owned())
+                .collect::<Vec<_>>()
         })
```
[the remark this answers is not in the document]
Comment (breferrari) [reply crates/vigia/src/view.rs:490]: Fixed in 59617db. detabbed now returns a Cow and borrows when the paragraph holds no tab, so the common case allocates nothing.

File: crates/vigia/src/lib.rs [hunk 3]
```diff
@@ -1216,6 +1218,25 @@ impl Shell {
...
+        let Some(store) = &self.store else {
+            return;
+        };
+        let refused: Vec<String> = ids
+            .iter()
+            .filter_map(|id| store.remove(id).err().map(|e| e.to_string()))
+            .collect();
+        for why in refused {
+            self.say(format!("could not remove a note: {why}"), Voice::Alert, now);
+        }
```
[the remark this answers is not in the document]
Comment (breferrari) [reply crates/vigia/src/lib.rs:1201]: Not changing this one, because the id is already in the message. Store::remove errors carry the path they failed on, and Error::Store renders as "could not use the notes store at <dir>/<id>.note: <source>", so the alert names the file and therefore the note. Adding the id to the prefix as well would print it twice on a footer that is width-constrained. If a bad id is what failed, path_of's error names it directly instead.
